pyanaconda/ui/gui/spokes/network.py

```diff
@@ -1526,16 +1526,13 @@ def apply(self):
 
         if self.networking_changed:
             if ANACONDA_ENVIRON in anaconda_flags.environs and self.payload.needsNetwork:
-                log.debug("network spoke (apply) refresh payload")
                 from pyanaconda.packaging import payloadMgr
-                if not self.payload.verifyAvailableRepositories():
-                    payloadMgr.restartThread(self.storage, self.data, self.payload, self.instclass,
-                                             fallback=not anaconda_flags.automatedInstall)
-                else:
-                    log.debug("Payload isn't restarted, repositories are still available.")
+                payloadMgr.restartThread(self.storage, self.data, self.payload, self.instclass,
+                                         fallback=not anaconda_flags.automatedInstall, onlyOnChange=True)
+                self.networking_changed = False
             else:
-                log.debug("network spoke (apply), payload refresh skipped (running outside of installation environment)")
-            self.networking_changed = False
+                log.debug("network spoke (apply), payload refresh skipped "
+                          "(running outside of installation environment)")
```

Comment:
A small nitpick, the message is not quite correct - and was not before, but while on that we could fix it.
The thing is that the restart can be skipped also in case of self.payload.needsNetwork == False and the log message will be still written out.

Also, shouldn't we unset networking_changed (as before the patch) also if the restart is not needed based on the condition(s) so we don't think that networking has changed next time we visit the spoke ?
I don't know, maybe move the needsNetwork condition one new if level up?
```
if self.networking_changed:
    if self.payload.needsNetwork:
        if ANACONDA_ENVIRON ...
            ...
        else:
            log.debug(... the same message ...)
    self.networking_changed = False
```

Reply:
True and true. I'm going to create new patch for this.